packages/lit-dev-content/samples/tutorials/intro-to-lit/11ty-data.cjs.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * This file MUST be CJS to be consumed by 11ty's data system.
+ *
+ * These files must be in the root of the tutorial directory and called
+ * 11ty-data.cjs.(js|ts).
+ *
+ * They must also export a getData async function and getManifest async fn.
+ *
+ * Upon creation of one of these files:
+ *
+ * 1. You must create a set of matching tutorial descriptions at
+ *    `/site/tutorials/content/[tutorial-directory-name]/[slug].md`
+ *
+ * 2. Do a call to `await loadTutorialData('[tutorial-directory-name]')` in
+ *    the order that you want it to appear in
+ *    `/site/tutorials/tutorials.11tydata.js`
+ */
+import {TutorialData, TutorialManifest} from '../utils.cjs.js';
+import path from 'path';
+
+/**
+ * Generates a tutorial data object for this tutorial's catalog card. For more
+ * options, see the TutorialData type in /samples/tutorials/utils.cjs.ts
+ *
+ * @returns Tutorial Data Object that describes this tutorial for the catalog.
+ */
+export const getData = async (): Promise<TutorialData> => {
```

Comment:
Could this be a JSON file, e.g. "lit-tutorial-metadata.json", instead of a node script? Seems like it would be simpler and more amenable to easily authoring tutorials (e.g. through the playground UI) in the future.

Not quite seeing why we need promises and the path library here. (Nothing is actually async, and the location seems like it could be inferred by whatever script is crawling through these directories.)

Reply:
I initially wrote this stuff as JSON but went with node because JSON format is really gross to deal with when authoring. 
* it's annoying to write multi-line strings for description
* you have to hunt down trailing commas
* you have to escape your quotes
* and usng TS:
  1. tells you if you are missing any required fields with type checking
  2. makes sure you're using the right types
  3. makes it easier to refactor the JSON format

I agree though that the filename probably isnt the best. What do you think I should change it to?

I can pull out the location as well and do that in the 11ty data file that consumes this. Can also remove the async. Idea behind it was that I was thinking i could use `fs` somehow to automate steps or decrease the number of project.jsons i'd have to write.

It's currently unnecessary, so would you still want me to remove the async?

Reply:
The problem with a JS file that I'm thinking of is that we wouldn't be able to safely evaluate it on the client. The use case I'm imagining is creating a tutorial as a gist in the playground editor. To me it seems natural to put metadata for that tutorial in a file like "manifest.json" or something. We could then read all the data we need out of that on the client side when you point the tutorial viewer at that gist. If it's a .js file, we would have to `eval` it, which would be unsafe.